javascript/javascript1/week2/flight.js

```diff
@@ -0,0 +1,142 @@
+      }else if (temp <=0) {
+        return " Better to wear jacket and scarf";
+      }else if (temp <= 15) {
+        return " A coat and shoes";
+      }else if (temp >=16 && temp <= 35) {
+        return "It's good weather outside, enjoy with short and t-shirt";
+      }
+    }
+    let clothesToWear = getYourClothes(25);
+    console.log(clothesToWear);
+
+    //Student manager
+
+    const class075Students = [];
+    function addStudentToClass(studentName) {
+      if (studentName === "") {
+        return "Please enter the student name";
+      }else {
+        if (class075Students.includes(studentName)!== true) {
+          if (class075Students.length<6) {
+            class075Students.push(studentName);
+          }else if (class075Students.length >=6 && studentName === "Queen") {
+            class075Students.push(studentName);
+          }else {
+            console.log("Cannot add more students to class 07");
+          }
+        }else {
+          console.log("Student" + studentName + "is already in the class");
+        }
+      }
+    }
+    function getNumberOfStudents() {
+      return class075Students.length;
+    }
+    addStudentToClass("Venkat");
+    addStudentToClass("Johnny");
+    addStudentToClass("Robert");
+    addStudentToClass("Rocky");
+    addStudentToClass("");
+    addStudentToClass("Vishnu");
+    addStudentToClass("Ahmad");
+    addStudentToClass("Raju");
+    addStudentToClass("Mohan");
+    addStudentToClass("Queen");
+    addStudentToClass("Ram");
+    console.log(class075Students);
+    console.log(getNumberOfStudents());
+
+    // Candy helper optional
+    boughtCandyPrices = [];
+    function addCandy(candyType, weight) {
+      if(candyType === "sweet") {
+        priceSweet = weight * 0.5;
+        boughtCandyPrices.push(priceSweet);
+      }else if (candyType === "chocolate") {
+          priceChocolate = weight * 0.7;
+          boughtCandyPrices.push(priceChocolate);
+        }else if (candyType === "toffee") {
+            priceToffee = weight * 1.1;
+            boughtCandyPrices.push(priceToffee);
+        }else if (candyType === "chewinggum") {
+            priceChewingGum = weight * 0.03;
+            boughtCandyPrices.push(priceChewingGum);
+          }
+        }
+      
+        addCandy("sweet", 30);
+        addCandy("chocolate", 20);
+        addCandy("toffee", 10);
+        addCandy("chewinggum", 5);
+
+        console.log(boughtCandyPrices);
+
+        const amountToSpend = Math.random() * 100;
+        let candyPrice = 0;
+        for(let i=0; i<boughtCandyPrices.length; i++) {
+          candyPrice += boughtCandyPrices[i];
+        }
+          console.log(candyPrice);
+          const totalprice = candyPrice;
+
+        function canBuyMoreCandy(totalprice) {
+          if(amountToSpend>candyPrice) {
+            console.log("You can buy more candy, so please do!");
+          }else {
+            console.log("Enough candy for you!");
+          }
+        }
```

Comment:
Although your function works, it can be a little dodgy. You have assigned the ```candyPrice``` to a new variable ```totalprice``` which in general is a good practice! Then you assign to the ```canBuyMoreCandy``` the parameter ```totalprice```  but you never use it 🤔 instead you compare the ```amountToSpend``` with the ```candyPrice``` which can be a little complicated. 
Try to assign a parameter to your function with a name that you have not already declared and then compare the ```amountToSpend``` with this parameter  😉

Reply:
done